### DISCUSSION TITLE
Errors looking for l.as

### DISCUSSION THREAD

**Discussion by JLMarin64** [moj-analytical-services/splink#196] [Q&A] [answered]
I'm getting an error where it is looking for a column "l.as" and I have no idea what that is or why it is looking for it.  It isn't part of my spark dataframe and I don't see that I need to add any additional columns other than unique_id or datasource. For this example, I'm only deduping and using only 1 spark dataframe as input so datasource was not provided in the columns.  Any clues as to what is going on here? 

![image](https://user-images.githubusercontent.com/15859431/113202774-3088e080-9239-11eb-9f2e-4f97ff152bec.png)

![image](https://user-images.githubusercontent.com/15859431/113202790-354d9480-9239-11eb-8639-60a6578968e6.png)

**Comment by RobinL** [comment, marked as answer]
Thanks for reporting this.  I think this is a bug, which I will try to track down properly.

For now, i suspect that if you rename your columns it will work.    I'm not sure which is causing the problem but i think it's most likely to be T_ADR_LN1.

If you try renaming:
D_BRTH -> dob
T_ADR_LN1 -> address
C_ZIP -> zip_code

I suspect it will probably work

**Reply by JLMarin64** [reply]
I renamed the columns like you suggested and it didn't work still.  I also renamed first and last name to the examples as well and it still didn't work.  Only other things I'm doing to this pyspark dataframe is reading in from s3 parquet file, I'm cleaning up data, filtering it, and renaming columns.  Once I'm done with that, I persist the pyspark df.  

![image](https://user-images.githubusercontent.com/15859431/113577805-d3ea4480-95ef-11eb-8df5-ad949b17fd07.png)

**Reply by RobinL** [reply]
Ah - I think i've got it.  

Additional columns to retain should either be omitted entirely, or be a blank array.  At the moment you have an array with a single value in it, `""`.  i.e. Splink it looking for a column named `""`

**Reply by JLMarin64** [reply]
It is running now. Thanks.
